### PR TITLE
Test a range of stages in CircularBufferingTest

### PR DESCRIPTION
**Pull request by rdspring1**
This PR adds a parameterized test suite for circular buffering. It is used to test a range of buffers from 2 to 4.

TODO: Add a runtime check to ensure that entire pipeline is filled by given inputs. See #2699.

### CODE REVIEW

**Comment by rdspring1** [comment]
!build

**Comment by naoyam** [comment]
Looks like the test failures are real failures?

**Comment by rdspring1** [comment]
!build

**Comment by rdspring1** [comment]
!build

**Comment by rdspring1** [comment]
!build

File: tests/cpp/test_loop_rotation.cpp [hunk 1]
```diff
@@ -648,11 +648,18 @@ __global__ void CUDAGeneratedKernel(Tensor<float, 2, 2> T0, Tensor<float, 2, 2>
          = T2[0LL];
     }
     NVFUSER_UPDATE_MAGIC_ZERO;
-    float T2[1LL];
-    T2[0LL]
-       = T1[0LL];
-    T3[(2LL + i13)]
-       = T2[0LL];
+    #pragma unroll
```
Comment (naoyam) [inline tests/cpp/test_loop_rotation.cpp:651]: Did you forget reverting this change?

**Review by naoyam — APPROVED** [review]
I think you forgot reverting one change in the loop rotation test. Otherwise looks good.

**Comment by rdspring1** [comment]
!build

**Comment by rdspring1** [comment]
The `H100` failures are:
```
[  FAILED  ] CombineMulSumAsMmaTestWithLayout.UseMatmulScheduler/NT, where GetParam() = 4-byte object <00-00 00-00>
[  FAILED  ] CombineMulSumAsMmaTestWithLayout.UseMatmulScheduler/NN, where GetParam() = 4-byte object <03-00 00-00>
```
Both occur on ToT main and are unrelated to this PR.

**Pull request merged**